Refactor `enroll-resources/agents`

ptgott opened the pull request:
Give the section a more reasonable information architecture. The section includes two children, one of which is a subsection (Joining Services) and another is an introductory overview.

Move `agents/introduction.mdx` to make it the new `agents/agents.mdx` section index page. This page is a high-level introduction, so it makes more sense as the section index. Also edit the page to:

- Remove an unnecessary list of links to sections for each agent service.
- Condense the architecture overview H2 to pave the way for making this page more of a section intro and less of an intro/conceptual hybrid guide.

Move "Join services" guides to make them direct descendents of the `agents` directory. This way, the substantial content of the section can have a higher directory level.

Extract the information on adding a service to an existing agent into a full how-to guide. This makes the instructions more visible. Also include example snippets within the guide.

ptgott committed:
Refactor `enroll-resources/agents`

Give the section a more reasonable information architecture. The section
includes two children, one of which is a subsection (Joining Services)
and another is an introductory overview.

Move `agents/introduction.mdx` to make it the new `agents/agents.mdx`
section index page. This page is a high-level introduction, so it makes
more sense as the section index. Also edit the page to:

- Remove an unnecessary list of links to sections for each agent
  service.
- Condense the architecture overview H2 to pave the way for making this
  page more of a section intro and less of an intro/conceptual hybrid
  guide.

Move "Join services" guides to make them direct descendents of the
`agents` directory. This way, the substantial content of the section can
have a higher directory level.

Extract the information on adding a service to an existing agent into a
full how-to guide. This makes the instructions more visible. Also
include example snippets within the guide.

ptgott commented:
Note that there are a lot of files with small changes to update link paths.

mmcallister commented on docs/pages/enroll-resources/agents/add-service-to-agent.mdx:
```diff
@@ -0,0 +1,264 @@
+---
+title: Enable a New Service on an Agent
+description: Explains how to edit the services taht are running on a Teleport Agent.
```
```suggestion
description: Explains how to edit the services that are running on a Teleport Agent.
```

mmcallister commented on docs/pages/enroll-resources/agents/add-service-to-agent.mdx:
```diff
@@ -0,0 +1,264 @@
+
+See the `teleport-kube-agent` [chart
+reference](../../reference/helm-reference/teleport-kube-agent.mdx#roles) for the
+roles and token tyeps that the chart supports.
```
```suggestion
roles and token types that the chart supports.
```

mmcallister commented on docs/pages/enroll-resources/agents/add-service-to-agent.mdx:
```diff
@@ -0,0 +1,264 @@
+## Further reading: manage tokens as code
+
+While this guide shows you how to create a token using `tctl`, we recommend that
+admins of production clusters manage tokens using the Teleport Terraform
```
Is there a bit of context that could be added to the recommendation here, as to why it would be better to use one of those other processes instead of `tctl` as this guide shows? Is this just better for one-off services for example?

ptgott replied:
I've softened up the language to say using the Terraform provider and Kubernetes operator are alternative possibilities. This way, users can decide for themselves which approach makes more sense for them!

ptgott committed:
Respond to mmcallister feedback

- Fix spelling.
- Soften the language about recommending the Terraform provider and
  Kubernetes operator so we don't need to justify a recommendation.
  Instead, let readers decide for themselves.

stevenGravy commented on docs/pages/enroll-resources/agents/add-service-to-agent.mdx:
```diff
@@ -0,0 +1,263 @@
+
+Generate a new join token for all services running on an agent, including any
+new services you want to run. In our example, we want to create a join token for
+the Teleport Kubernetes Service and Teleport Application Service, so we specify
```
it's not entirely clear here which service was the existing and which one we're adding. I'd make that more clear.

ptgott committed:
Respond to stevenGravy feedback

Clarify the need to create a token for an additional service you want to
run.

ptgott committed:
Fix redirects
